pkg/pxc/app/statefulset/node.go

```diff
@@ -265,6 +267,52 @@ func (c *Node) AppContainer(spec *api.PodSpec, secrets string, cr *api.PerconaXt
 		}...)
 	}
 
+	if cr.CompareVersionWith("1.18.0") >= 0 {
+		const LD_PRELOAD_KEY = "LD_PRELOAD"
+		const LIBJEMALLOC_PATH = "/usr/lib64/libjemalloc.so.1"
+		activateJemalloc := false
+		ldPreloadValue := ""
+
+		if cr.Spec.PXC.MySQLAllocator == "" || strings.ToLower(cr.Spec.PXC.MySQLAllocator) == "jemalloc" {
+			activateJemalloc = true
+		}
+
+		// Let's set LD_PRELOAD via appc.Env always. It takes precedence over EnvFrom
+
+		envVarsSecret := &corev1.Secret{}
+		err := cl.Get(ctx, types.NamespacedName{Name: cr.Spec.PXC.EnvVarsSecretName, Namespace: cr.Namespace}, envVarsSecret)
+		if client.IgnoreNotFound(err) == nil {
+			// Env vars are set via secret. Check if LD_PRELOAD is set.
+			for key, value := range envVarsSecret.Data {
+				if key == LD_PRELOAD_KEY {
+					ldPreloadValue = string(value)
+					break
+				}
+			}
```

Comment:
We can avoid this loop completely by using 

```
if val, ok := envSecret.Data[LdPreloadKey]; ok {
    ldPreload = string(val)
}
```

Data of secret are of type `map[string][]byte `.